plugins/contiv/host.go

```diff
@@ -292,3 +309,29 @@ func (s *remoteCNIserver) ipPrefixToAddress(ip string) string {
 	}
 	return ip
 }
+
+func (s *remoteCNIserver) getHostLinkIPs() ([]string, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		s.Logger.Error("Unable to list host links:", err)
+		return nil, err
+	}
+
+	res := make([]string, 0)
+	for _, l := range links {
+		if !strings.HasPrefix(l.Attrs().Name, "lo") && !strings.HasPrefix(l.Attrs().Name, "docker") &&
+			!strings.HasPrefix(l.Attrs().Name, "virbr") && !strings.HasPrefix(l.Attrs().Name, "vpp") {
+			// not a virtual interface, list its IP addresses
+			addrList, err := netlink.AddrList(l, netlink.FAMILY_V4)
+			if err != nil {
+				s.Logger.Error("Unable to list link IPs:", err)
+				return nil, err
+			}
+			// return all IPs
+			for _, addr := range addrList {
+				res = append(res, addr.IP.String())
+			}
+		}
+	}
+	return res, nil
+}
```

Comment:
is there any way to detect if host IPs change (CCP primary interface is DHCP for example, or maybe a down interface comes up) to make route adjustments?

An error might be good too if a host IP overlaps one of the IPAM networks?

Reply:
This is pretty-much a temporary solution (ETA to be removed is 1 week). To fix another issue, we need to ask k8s what k8s thinks is the main IP address of our node. This will be done via k8s state reflector (therefore needs some more time to implement). I added a TODO explaining this. I think that addressing these corner cases is not worth of time investment at this point.

Reply:
yeah getting from node internal ip is more reasonable